### DISCUSSION TITLE
Support for fifo and commit-timing protocols

### DISCUSSION THREAD

**Discussion by danielfikko** [labwc/labwc#2480] [feature-requests]
Wayland has two new protocols: [fifo](https://gitlab.freedesktop.org/wayland/wayland-protocols/-/blob/main/staging/fifo/fifo-v1.xml) and [commit-timing](https://gitlab.freedesktop.org/wayland/wayland-protocols/-/blob/main/staging/commit-timing/commit-timing-v1.xml). Both are essential for a good [gaming](https://github.com/libsdl-org/SDL/pull/9383) and [multimedia](https://github.com/mpv-player/mpv/pull/15056) experience in wayland.

There is already a pending implementation of both protocols on the wlroots side [[1]](https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/4463) [[2]](https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/4617), but i suppose it also needs a small implementation by labwc, just like [sway](https://github.com/swaywm/sway/pull/8371).

So it would be great if labwc supported such protocols.

**Comment by Consolatis** [comment]
That requires the MRs to be merged on the wlroots side, a new wlroots release which includes those MRs and labwc tracking that release. So it will take a while.

**Reply by danielfikko** [reply]
I see.

Just so I don't create another two discussions:

In wlroots support for the `xdg-toplevel-icon`, `ext-image-capture-source` and `ext-image-copy` protocols has just been merged [[1]](https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/4704) [[2]](https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/4545). Do you intend to support these protocols?

These are quite exciting protocols, because `xdg-toplevel-icon` allows programs running on [wine](https://gitlab.winehq.org/wine/wine/-/merge_requests/6938) to have an icon in the taskbar and `ext-image-capture-source`/`ext-image-copy` allows you to share just single window instead of the entire output.

**Reply by Consolatis** [reply]
Yes, labwc intends to support both of these protocols once we start tracking wlroots 0.19. Note that the linked wlroots MR does not implement a toplevel capture source but only a output capture source.
